src/viam/sdk/common/utils.cpp

```diff
@@ -25,24 +25,26 @@ using viam::common::v1::ResourceName;
 
 std::vector<ResourceName> resource_names_for_resource(const std::shared_ptr<Resource>& resource) {
     std::string resource_type;
+    std::string resource_subtype;
     std::vector<ResourceName> resource_names;
     for (auto& kv : Registry::registered_models()) {
         const std::shared_ptr<ModelRegistration> reg = kv.second;
         if (reg->api().to_string() == resource->dynamic_api().to_string()) {
-            resource_type = reg->api().resource_subtype();
+            resource_type = reg->api().resource_type();
+            resource_subtype = reg->api().resource_subtype();
         } else {
             continue;
         }
 
-        if (resource_type.empty()) {
-            resource_type = resource->name();
+        if (resource_subtype.empty()) {
+            resource_subtype = resource->name();
         }
 
         ResourceName r;
         *r.mutable_namespace_() = kRDK;
-        *r.mutable_type() = resource->type().to_string();
         *r.mutable_name() = resource->name();
-        *r.mutable_subtype() = resource_type;
+        *r.mutable_type() = resource_type;
+        *r.mutable_subtype() = resource_subtype;
```

Comment:
@benjirewis - These are the lines I was referring to in my comment.

Reply:
Ah I see. You needed to change it because `resource->type()` is no longer a thing. At first I didn't see that and I couldn't see what it was that required changes here due to the removal of ResourceType.

I do agree that the code is pretty funky. Hopefully @stuqdog can clarify whether what you have here is OK or needs the handling you suggest, as I'm not certain either.

Reply:
Sorry for the slow response. I think what we have here is correct. In the event that `(reg->api().to_string() != resource->dynamic_api().to_string())` then the registered model we're looking at is not of the provided resource type and so we `continue` on line 34. This isn't a case where we're creating a `ResourceName` that lacks a `resource_type` but rather a case where we aren't creating a `ResourceName` at all.

Reply:
Ah cool that makes sense! Thanks for context.